dali/python/nvidia/dali/ops/__init__.py

```diff
@@ -411,6 +415,23 @@ def _process_instance_name(self, arguments):
         else:
             self._name = "__" + type(self._op).__name__ + "_" + str(self._counter.id)
 
+    def _process_trace(self, arguments):
+        if _dali_trace.is_tracing_enabled():
+            if _Pipeline.current():
+                skip_bottom = _Pipeline.current()._definition_stack_frame
+            else:
+                skip_bottom = 0
+            skip_top = 5 if self._op._api == "fn" else 3
```

Comment:
I was thinking about other solutions: 
* I got the suggestion to use decorator, that would adjust the API functions with stop marker - but I only deal with the frames in the for of `FrameSummary` that contain: filename, the line number, the function name and the code context (as a string) - I cannot look up runtime metadata hera without going through the actual stack frames and this will probably be much heavier.
* Filtering based on the function names also ties with the implementation, I would say even more than just the magic number.
If we change something in the internal implementation, the length of the stack trace will change and the test will fail.
I am also not super happy about the magic numbers, but I saw it in TF as well - when you want to get the stack trace for something in your API, and you put the code in the helper function you change the contents of the stack trace :(

Reply:
For going through actuall stack frames, I think the same, it will be likely much heavier and kind of easier to make it wrong (e.g. to introduce some circular references that would bloat up memory usage).
For filtering by name, sure, it's still tying to the implementation, I just feel that being explicit here is a benefit for maintainers. If you think otherwise, how about a comment listing exact functions we expect to remove with those magic numbers? I am mostly conerned with the part that looks back and makes assumption of how many calls have been made before entering the function that has the magic constant defined.

Or maybe, on the explicity part, the front-end operator constructors could just create some stack_context (as simple as an integer - the stack size) and just painstakingly pass it through the utlities to the point where we acquire the stack. When making adjustments to the code in the future, you just need to remember to pass it and it works.

Reply:
Done (I hope).